PowerSync/PowerSync.Common/Client/PowerSyncDatabase.cs

```diff
@@ -373,11 +380,26 @@ await Database.WriteTransaction(async tx =>
         Emit(new PowerSyncDBEvent { StatusChanged = CurrentStatus });
     }
 
-    public new async Task Close()
+    /// <summary>
+    /// Close the database, releasing resources.
+    ///
+    /// Also disconnects any active connection.
+    /// 
+    /// Once close is called, this connection cannot be used again - a new one
+    /// must be constructed.
+    /// </summary>
+    public async Task Close(bool disconnect = true)
```

Comment:
For .Net I think we probably should always `disconnect`, at-least initially. I believe web has this setting for multiple tab configurations.